mailpoet/mailpoet
# Redesign the MSS step of the welcome wizard [MAILPOET-4818]

## Description

This PR implements the new MSS step of the Welcome Wizard. It is a big PR as it involves design and functional changes to the MSS step, some JSX to TSX refactoring, and refactoring of the code to validate an MSS key so that it can be used both in the settings page and in the wizard. Check the Jira ticker for more details on the specific changes.

## Code review notes

@samnajian already helped with a review of a draft version of this PR and helped me confirm a few of the decisions that I made when implementing this PR. But he did not check the most recent commits, and anyway, I believe a full review is valid. I would suggest reviewing this PR commit by commit. I tried my best to make the commits granular and the commit messages relevant to make the reviewing process easier.

I left one inline comment with a detail that I'm not sure about.

## QA notes

This PR makes significant changes to both the welcome wizard and the "Key Activation" tab of the settings page. The only visible change in the "Key Activation" tab is that now the "Verify" button is disabled when there is no key. The other changes were refactors without any visible change to the user.

Please test both pages carefully as there are a lot of code changes in this PR.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4818]

## After-merge notes

_N/A_


[MAILPOET-4818]: https://mailpoet.atlassian.net/browse/MAILPOET-4818?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
